include/swift/AST/LifetimeDependence.h

```diff
@@ -0,0 +1,110 @@
+//===--- LifetimeDependenceSpecifiers.h ------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines types and utilities related to Lifetime Dependence
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_AST_LIFETIMEDEPENDENCE_H
+#define SWIFT_AST_LIFETIMEDEPENDENCE_H
+
+#include "swift/AST/Decl.h"
+#include "swift/Basic/Debug.h"
+#include "swift/Basic/OptionSet.h"
+#include "llvm/Support/TrailingObjects.h"
+
+namespace swift {
+
+enum class LifetimeDependenceKind : uint8_t {
+  Copy = 0,
+  Consume = 1,
```

Comment:
Nit: you don't actually need to specify each value, increments are handled automatically starting from initial value.